### PR TITLE
Add code coverage support

### PR DESCRIPTION
**Pull request by andreas-bok-sociomantic**
- Add autoconf macro to configure code coverage 
- Integrate code coverage generation and upload to codecov.io to the continuous integration on Travis

### CODE REVIEW

**Comment by bokchan** [comment]
Rebased on latest v6.x.x branch

**Comment by bokchan** [comment]
Reviews welcome if you are anyway bored on your Christmas holiday 😂

**Comment by bokchan** [comment]
@leandro-lucarella-sociomantic 
Is it possible to add me, i.e. `@bokchan` as admin for https://codecov.io/gh/sociomantic-tsunami/libdrizzle-redux ? I assume I will still be maintaining the library unless you found an internal person to take over.

File: m4/ax_code_coverage.m4 [hunk 1]
```diff
@@ -61,6 +61,7 @@
 #   Copyright (c) 2012 Paolo Borelli
 #   Copyright (c) 2012 Dan Winship
 #   Copyright (c) 2015 Bastien ROUCARIES
+# 	Copyright (c) 2018 sociomantic labs GmbH
```
Comment (ben-palmer-sociomantic) [inline m4/ax_code_coverage.m4]: Should be Dunnhumby GMBH now I think.

**Review by ben-palmer-sociomantic — COMMENTED** [review]
Made one minor comment. LGTM otherwise but I don't know much about the build system.

File: m4/ax_code_coverage.m4 (hunk 1 above)
Comment (bokchan) [reply m4/ax_code_coverage.m4]: True

**Comment by bokchan** [comment]
Fixed copyright notice in `ax_code_coverage.m4`

**Comment by bokchan** [comment]
Feel free to merge if you are satisfied, @ben-palmer-sociomantic  
Do you want to test an v6.2.0 alpha before the official release?

**Comment by bokchan** [comment]
?

**Comment by bokchan** [comment]
rebased on latest v6.x.x after v6.1.6 release

**Comment by ben-palmer-sociomantic** [comment]
LGTM

**Pull request merged**